dev.zed.Zed.Preview.yaml

```diff
@@ -0,0 +1,210 @@
+      - install -Dm 644 ${FLATPAK_ID}.metainfo.xml --target-directory ${FLATPAK_DEST}/share/metainfo
+      - install -Dm 644 share/icons/hicolor/512x512/apps/zed.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/zed-preview.png
+
+      # Rename instances of `zed` to `${FLATPAK_ID}`
+      - rename zed-preview ${FLATPAK_ID} ${FLATPAK_DEST}/share/{applications/*,icons/hicolor/*/apps/*}
+
+      # Ensure `desktop-file-edit` validates; Enforce `[Desktop Action NewWorkspace]` has a matching action
+      - |
+        if ! grep -q "^Actions=NewWorkspace" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"; then
+          # Find the line number where [Desktop Action NewWorkspace] is located
+          line_num=$(grep -n "^\[Desktop Action NewWorkspace\]" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop" | cut -d: -f1)
+          if [ -n "$line_num" ]; then
+            # Insert Actions=NewWorkspace above the line where [Desktop Action NewWorkspace] is located
+            sed -i "${line_num}i Actions=NewWorkspace" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+          else
+            # If [Desktop Action NewWorkspace] doesn't exist, add Actions=NewWorkspace at the end
+            echo "Actions=NewWorkspace" >> "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+          fi
+        fi
+
+      # Ensure zed-wrapper is used as the Exec command in the `.desktop` file
+      - desktop-file-edit --set-key="Exec" --set-value="zed-wrapper %U" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      # Modify original `.desktop` file to use the correct icon name
+      - desktop-file-edit --set-icon="${FLATPAK_ID}" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+    sources:
+      - type: archive
+        url: https://github.com/zed-industries/zed/releases/download/v0.184.4-pre/zed-linux-x86_64.tar.gz
+        sha256: 16f51a57dda2f05fd1e17e370c0447446e6afb05c565f61f85c9f309c8f4f134
+        only-arches:
+          - x86_64
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/zed-industries/zed/releases
+          version-query: map(select(.prerelease)) | first | .tag_name
+          url-query:
+            map(select(.prerelease)) | first | .assets[] | select(.name=="zed-linux-x86_64.tar.gz")
+            | .browser_download_url
+          is-main-source: true
+      - type: archive
+        url: https://github.com/zed-industries/zed/releases/download/v0.184.4-pre/zed-linux-aarch64.tar.gz
+        sha256: fdaf297297bdcc12859364d2f0303666565d7d2b818dd738e63cf1373f227178
+        only-arches:
+          - aarch64
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/zed-industries/zed/releases
+          version-query: map(select(.prerelease)) | first | .tag_name
+          url-query:
+            map(select(.prerelease)) | first | .assets[] | select(.name=="zed-linux-aarch64.tar.gz")
+            | .browser_download_url
```

Comment:
```suggestion
          url: https://api.github.com/repos/zed-industries/zed/releases
          version-query: '[.[] | select(.prerelease)][0].tag_name'
          url-query: '[.[] | select(.prerelease)][0].assets[] | select(.name=="zed-linux-aarch64.tar.gz")
            | .browser_download_url'
```

Reply:
this does the same as before...

Reply:
For some reason I don't know why I got an error when I generated the package with this code.

Reply:
Okay, I can update that then. Did a build work locally?

Reply:
Skip this - now working fine, was a bug in JQ

Reply:
Sure. Did the app build locally with your manifest?

Reply:
Yes, I make some changes - I'm looking at doing a PR on your branch.

Reply:
Done https://github.com/vixalien/flathub/pull/1
```
flatpak run org.flatpak.Builder .flatpak dev.zed.Zed-Preview.yaml --force-clean --install --user
flatpak run --user dev.zed.Zed-Preview
flatpak uninstall --user dev.zed.Zed-Preview 

```